service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalogHandler.java

```diff
@@ -580,7 +586,18 @@ public Optional<LoadTableResponse> loadTableIfStale(
       }
     }
 
-    return Optional.of(CatalogHandlers.loadTable(baseCatalog, tableIdentifier));
+    Table table = baseCatalog.loadTable(tableIdentifier);
+    if (table instanceof BaseMetadataTable) {
+      throw new NoSuchTableException("Table does not exist: %s", tableIdentifier.toString());
+    } else if (!(table instanceof BaseTable)) {
+      throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
+    } else {
+      LoadTableResponse rawResponse =
+          LoadTableResponse.builder()
+              .withTableMetadata(((BaseTable) table).operations().current())
+              .build();
+      return Optional.of(filterResponseToSnapshots(rawResponse, snapshots));
+    }
```

Comment:
Couldn't you keep the call to `CatalogHandlers.loadTable`?

```suggestion
    LoadTableResponse rawResponse = CatalogHandlers.loadTable(baseCatalog, tableIdentifier);
    return Optional.of(filterResponseToSnapshots(rawResponse, snapshots));
```